### ISSUE TITLE
Allow unknown toolsets in SociConfig.cmake

### ISSUE THREAD

**Issue by mloskot** [SOCI/soci#104]
From recent thead on [CMake FInd threads - iOS](http://sourceforge.net/mailarchive/message.php?msg_id=30602608) that spawned to CMake mailing list too, 

```
cmake /Prog/Frameworks/soci/src/ -DCMAKE_IOS_SDK_ROOT=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS6.1.sdk -DCMAKE_TOOLCHAIN_FILE=/Prog/iOS.cmake -GXcode -DWITH_BOOST=OFF
 -- Configuring SOCI:
 -- SOCI_VERSION                             = 3.2.0
 -- SOCI_ABI_VERSION                         = 3.2
 CMake Error at cmake/SociConfig.cmake:50 (message):
 CMake is unable to recognize compilation toolset to build SOCI for you!
 Call Stack (most recent call first):
 CMakeLists.txt:41 (include)
```

As @snikulov pointed in the thread, we should relax the FATAL_ERROR from SociConfig.cmake

**Comment by snikulov** [comment]
@mloskot Can we close this?

**Comment by mloskot** [comment]
Yes, it has been fixed in master, in 1d26033e060c2108d4ffbe06bbe4a6895c24bdc7
